server/src/main/java/io/crate/metadata/cluster/DDLClusterStateHelpers.java

```diff
@@ -68,9 +68,8 @@ public static IndexTemplateMetadata updateTemplate(IndexTemplateMetadata indexTe
             // merge and validate settings (private settings must be filtered out, they are handled internally)
             Settings mergedSettings = Settings.builder()
                 .put(indexTemplateMetadata.settings())
-                .put(newSettings)
-                .build()
-                .filter(k -> indexScopedSettings.isPrivateSetting(k) == false);
+                .put(newSettings.filter(k -> indexScopedSettings.isPrivateSetting(k) == false))
+                .build();
```

Comment:
Originally the filter was applied to even existing template settings which caused the `version.created` to be removed. Now narrowing the filter to the new settings only.